vaadin-select-flow-parent/vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java

```diff
@@ -449,6 +451,7 @@ public void setItems(Stream<T> streamOfItems) {
     @Deprecated
     public void setDataProvider(DataProvider<T, ?> dataProvider) {
         this.dataProvider.set(dataProvider);
+        DataViewUtils.removeComponentFilterAndSortComparator(this);
```

Comment:
There should be a unit test for Select as I've done the integration originally. Please verify this there

Reply:
Validated in `AbstractListDataViewListenerTest`